### PR TITLE
replace Net::HTTP with HTTPClient

### PR DESCRIPTION
**Pull request by takkanm**
part or #36

### CODE REVIEW

**Comment by nahi** [comment]
LGTM, but since this underlying change needs overall regression test I want to wait GET implementation merged. Let's release other fixes first then merge this for the next release?

**Comment by takkanm** [comment]
@nahi I replaced GET implementation too.

File: lib/td/client/api.rb [hunk 1]
```diff
@@ -392,27 +328,22 @@ def do_get(url, params=nil, &block)
             infl.close
           end
         else
-          body = Zlib::Inflate.inflate(body)
+          begin
+            # NOTE maybe for content-encoding is msgpack.gz ?
+            body = Zlib::Inflate.inflate(body)
+          rescue
+            # NOOP
```
Comment (nahi) [inline lib/td/client/api.rb]: How this happens?

File: lib/td/client/api.rb [hunk 2]
```diff
@@ -392,27 +328,22 @@ def do_get(url, params=nil, &block)
...
+          end
         end
       end
     end
 
-    return [response.code, body, response]
+    return [response.code.to_s, body, response]
```
Comment (nahi) [inline lib/td/client/api.rb]: Is to_s necessary? Could you check if callees expect Integer or String?

File: lib/td/client/api/job.rb [hunk 3]
```diff
@@ -201,29 +197,29 @@ def job_result_each(job_id, &block)
...
+
+    get("/v3/job/result/#{e job_id}", {'format'=>'msgpack'}) {|res, chunk, current_total_chunk_size|
+      if res.code != 200
         raise_error("Get job result failed", res)
       end
 
-      res.extend(DirectReadBodyMixin)
-      if res.header['Content-Encoding'] == 'gzip'
+      # default to decompressing the response since format is fixed to 'msgpack'
+      if !infl && (res.header['Content-Encoding'].include? 'gzip')
```
Comment (nahi) [inline lib/td/client/api/job.rb]: What's the intention of this "!infl" ?

Comment (takkanm) [inline lib/td/client/api/job.rb]: This is a check for the second and subsequent loops don't want to create a Zlib:Inflate instance.

Comment (nahi) [inline lib/td/client/api/job.rb]: But if !infl == false it fallbacks to 'infl = Zlib::Inflate.new', no?

File: lib/td/client/api.rb (hunk 1 above)
Comment (takkanm) [inline lib/td/client/api.rb]: This code was not assume other than the Content-Encoding is 'gzip' or 'msgpack.gz'.
However, it has been modified so as not enter other than them.

https://github.com/treasure-data/td-client-ruby/commit/ab8e45c422b99c57175cff100b9ae7d285814a38

File: lib/td/client/api/job.rb (hunk 3 above)
Comment (takkanm) [inline lib/td/client/api/job.rb]: yes. This is what I will be rewritten as follows.

```
infl ||= Zlib::Infrate.new
```

now, I fixed it.

https://github.com/treasure-data/td-client-ruby/commit/aedd61c66472c256b4498a0b64059c13dbbe77da

**Comment by takkanm** [comment]
I was sure that the following items.

- some commands result
- http header
- speed of `td job:show -f csv`

Command results and http header did not change.
But, speed is Speed had become slower as the 4%.

**Comment by nahi** [comment]
LGTM!

**Pull request merged**